automat/_test/test_visualize.py

```diff
@@ -1,16 +1,34 @@
-
-from __future__ import unicode_literals
+import functools
 
 import os
 import subprocess
 from unittest import TestCase, skipIf
 
+from characteristic import attributes
+
 from .._methodical import MethodicalMachine
 
+from .._visualize import elementMaker, tableMaker
+
+
+def isGraphvizModuleInstalled():
+    """
+    Is the graphviz Python module installed?
+    """
+    try:
+        import graphviz
+    except ImportError:
+        return False
+    else:
+        del graphviz
```

Comment:
What I'd really like is a way to fool branch coverage into thinking that
the `ImportError` always gets caught.  The straight forward way is to add a
new tox test environment that excludes all optional dependencies and thus
always runs the skip logic. This is probably not worth the effort. ;)

On Mon, May 23, 2016 at 3:23 PM, Glyph <notifications@github.com> wrote:

> In automat/_test/test_visualize.py
> <https://github.com/glyph/automat/pull/20#discussion_r64299627>:
>
> >  from .._methodical import MethodicalMachine
> >
> > +from .._visualize import elementMaker, tableMaker
> > +
> > +
> > +def isGraphvizModuleInstalled():
> > +    """
> > +    Is the graphviz Python module installed?
> > +    """
> > +    try:
> > +        import graphviz
> > +    except ImportError:
> > +        return False
> > +    else:
> > +        del graphviz
>
> If you really feel strongly about it though you can always return graphviz
> itself; that's truthy :).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/glyph/automat/pull/20/files/1e34c3d7a06a8d9685168c94abdf3b77b4c694b5#r64299627>
>